src/test/java/org/apache/sling/xss/impl/AntiSamyPolicyWithTestConfigTest.java

```diff
@@ -0,0 +1,146 @@
+ ~ software distributed under the License is distributed on an
+ ~ "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ ~ KIND, either express or implied.  See the License for the
+ ~ specific language governing permissions and limitations
+ ~ under the License.
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~*/
+package org.apache.sling.xss.impl;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import javax.xml.stream.XMLStreamException;
+
+import java.io.IOException;
+import java.util.regex.Pattern;
+
+import org.apache.sling.xss.impl.xml.AntiSamyPolicy;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.MethodSource;
+
+public class AntiSamyPolicyWithTestConfigTest {
+
+    public static final String POLICY_FILE = "./testConfig.xml";
+    private static HtmlSanitizer antiSamy;
+
+    @BeforeAll
+    public static void setup() throws InvalidConfigException, XMLStreamException, IOException {
+        antiSamy = new HtmlSanitizer(new AntiSamyPolicy(AntiSamyPolicyWithTestConfigTest.class.getClassLoader().getResourceAsStream(POLICY_FILE)));
+    }
+
+    @ParameterizedTest
+    @MethodSource("dataForDynamicAttributes")
+    public void testDynamicAttributes(TestInput testInput) throws Exception {
+         testInput.skipComparingInputWithOutput = false;
+         testInput.runCheck();
+    }
+
+    @ParameterizedTest
+    @MethodSource("dataForGlobalAttributes")
+    public void testGlobalAttributes(TestInput testInput) throws Exception {
+         testInput.skipComparingInputWithOutput = false;
+         testInput.runCheck();
+    }
+
+
+    static TestInput[] dataForGlobalAttributes() {
+        return new TestInput[]{
+                new TestInput("<p draggable=\"wrong\">This is a paragraph.</p>",
+                        "<p",false),
+                new TestInput("<p draggable=\"auto\">This is a paragraph.</p>",
+                        "<p", true),
+                new TestInput("<p draggable=\"true\">This is a draggable paragraph.</p>",
+                        "<p", true),
+                new TestInput("<p contenteditable=\"wrong\">This is a paragraph.</p>",
+                        "<p",false),
+                new TestInput("<p contenteditable=\"true\">This is a paragraph.</p>",
+                        "<p", true),
+        };
+    }
+
+     static TestInput[] dataForDynamicAttributes() {
+        return new TestInput[]{
+                new TestInput("<p data-test=\"test-purpose\">This is a paragraph.</p>",
+                       "<p data-test=\"test-purpose\">This is a paragraph.</p>", true),
+                new TestInput("<p data-test=\"test\">This is a paragraph.</p>",
+                       "<p data-test=\"test\">This is a paragraph.</p>", true),
+                };
+        }
+
+    private static class TestInput {
```

Comment:
I think this TestInput class already exists in the code, right? Please extract it to a shared class so it can be reused instead of duplicated.

Reply:
Resolved: [9f0dbc5](https://github.com/apache/sling-org-apache-sling-xss/pull/30/commits/9f0dbc5a30a90a3949c0dcd7f2bc3b2bac41b420)